analyzers/src/SonarAnalyzer.Common/AnalysisContext/SonarAnalysisContextBase.cs

```diff
@@ -47,6 +47,9 @@ public abstract class SonarAnalysisContextBase<TContext> : SonarAnalysisContextB
     public SonarAnalysisContext AnalysisContext { get; }
     public TContext Context { get; }
 
+    private readonly ConcurrentDictionary<SyntaxTree, bool> shouldAnalyzeTreeCache = new();
```

Comment:
I don't like the idea of a Dictionary that holds references to SyntaxTrees on an object, that may outlive the compilation. See also https://github.com/dotnet/roslyn-analyzers/blob/main/src/Microsoft.CodeAnalysis.Analyzers/Microsoft.CodeAnalysis.Analyzers.md#rs1008-avoid-storing-per-compilation-data-into-the-fields-of-a-diagnostic-analyzer for why this isn't a good idea.

Let's keep this PR as a reference, but we should try what I proposed here as well.